schemas/com.snowplowanalytics.snowplow.enrichments/api_request_enrichment_config/jsonschema/1-0-0

```diff
@@ -0,0 +1,167 @@
+{
+	"$schema": "http://json-schema.org/schema#",
+	"description": "Schema for API Request Enrichment configuration",
+	"self": {
+		"vendor": "com.snowplowanalytics.snowplow.enrichments",
+		"name": "api_request_enrichment_config",
+		"format": "jsonschema",
+		"version": "1-0-0"
+	},
+
+	"type": "object",
+	"properties": {
+		"vendor": {
+			"type": "string"
+		},
+		"name": {
+			"type": "string"
+		},
+		"enabled": {
+			"type": "boolean"
+		},
+
+		"parameters": {
+			"type": "object",
+			"properties": {
+				"inputs": {
+					"type": "array",
+					"items": {
+						"type": "object",
+						"properties": {
+							"key": {
+								"type": "string",
+								"pattern": "^[A-Za-z0-9_-]+$"
+							},
+							"pojo": {
+								"type": "object",
+								"properties": {
+									"field": {
+										"type": "string"
+									}
+								},
+								"additionalProperties": false
+							},
+							"json": {
+								"type": "object",
+								"properties": {
+									"field": {
+										"type": "string",
+										"enum": ["unstruct_event", "contexts", "derived_contexts"]
+									},
+								  "schemaCriterion": {
+									  "type": "string",
+									  "pattern": "^iglu:[a-zA-Z0-9-_.]+/[a-zA-Z0-9-_]+/[a-zA-Z0-9-_]+/[0-9]+-([0-9]+|\\*)-([0-9]+|\\*)$"
+								  },
+								  "jsonPath": {
+									  "type": "string"
+								  }
+								},
+								"additionalProperties": false
+							}
+						},
+						"additionalProperties": false,
+						"minProperties": 2,
+						"maxProperties": 2,
+						"required": ["key"]
+					}
+				},
+				"api": {
+					"type": "object",
+					"minProperties": 1,
+					"maxProperties": 1,
+					"properties": {
+						"http": {
+							"type": "object",
+							"properties": {
+								"method": {
+									"type": "string",
+									"enum": ["GET", "POST", "PUT"]
+								},
+								"uri": {
+									"type": "string"
+								},
+								"timeout": {
+									"type": "integer",
+									"minimum": 1,
+									"maximum": 60000
+								},
+								"authentication": {
+									"type": "object",
+									"properties": {
+										"httpBasic": {
+											"type": "object",
+											"properties": {
+												"username": {
+													"type": ["string", "null"],
```

Comment:
@alexanderdean just to clarify. Remind me what we decided about optional username/password? Basic auth is basically header with `base64(concatenate(username, ':', password))` so in our case `null` is equal to empty string. As it will be `concatenate(username, ':', '')`. So does it make sense to use `null` at all?